standard/expressions.md

```diff
@@ -2064,9 +2122,9 @@ A *this_access* is permitted only in the *block* of an instance constructor, an
 - When `this` is used in a *primary_expression* within an instance method or instance accessor of a class, it is classified as a value. The type of the value is the instance type ([§15.3.2](classes.md#1532-the-instance-type)) of the class within which the usage occurs, and the value is a reference to the object for which the method or accessor was invoked.
 - When `this` is used in a *primary_expression* within an instance constructor of a struct, it is classified as a variable. The type of the variable is the instance type ([§15.3.2](classes.md#1532-the-instance-type)) of the struct within which the usage occurs, and the variable represents the struct being constructed.
   - If the constructor declaration has no constructor initializer, the `this` variable behaves exactly the same as an `out` parameter of the struct type. In particular, this means that the variable shall be definitely assigned in every execution path of the instance constructor.
-  - Otherwise, the `this` variable behaves exactly the same as a `ref` parameter of the struct type. In particular, this means that the variable is considered initially assigned.
+  - Otherwise, the `this` variable behaves exactly the same as a `in` parameter of the struct type. In particular, this means that the variable is considered initially assigned, and can't be reassigned.
 - When `this` is used in a *primary_expression* within an instance method or instance accessor of a struct, it is classified as a variable. The type of the variable is the instance type ([§15.3.2](classes.md#1532-the-instance-type)) of the struct within which the usage occurs.
-  - If the method or accessor is not an iterator ([§15.14](classes.md#1514-iterators)) or async function ([§15.15](classes.md#1515-async-functions)), the `this` variable represents the struct for which the method or accessor was invoked, and behaves exactly the same as a `ref` parameter of the struct type.
+  - If the method or accessor is not an iterator ([§15.14](classes.md#1514-iterators)) or async function ([§15.15](classes.md#1515-async-functions)), the `this` variable represents the struct for which the method or accessor was invoked, and behaves exactly the same as a `in` parameter of the struct type.
```

Comment:
Again, that doesn't appear to be the case.

```csharp
class Program
{
    static void Main()
    {
        Test t = new Test(5);
        t.ChangeMe(10);
        Console.WriteLine(t.value); // Prints 10
    }
}

struct Test
{
    public int value;

    public Test(int value)
    {
        this.value = value;
    }

    public void ChangeMe(int value)
    {
        this = new Test(value);
    }
}
```

Reply:
Also agreed.

Reply:
Action 2023-04-19: As above, this is a struct vs readonly struct difference.